chapters/04_game_theoretic_model/03_methodology/main.tex

```diff
@@ -0,0 +1,1596 @@
+\(i \in \{A, B\}\).
+In other words the queueing systems could start blocking type \(2\) individuals
+either when there is \(1\) individual in node 1 or when there are \(2\)
+individuals.
+See Figure~\ref{fig:diagram_of_game_theoretic_model} for a visual representation
+of the game.
+In fact, in this example, the only difference between the two queueing systems
+is that queueing system \(A\) has a lower service rate than queueing system
+\(B\).
+
+As shown in Section~\ref{sec:queueing_systems_and_normal_form_games} the payoff
+matrices for the two players can be obtained using the function
+\lstinline[style=pystyle]{get_payoff_matrices}.
+Every function that described in this section is also implemented in the
+\lstinline[style=pystyle]{ambulance_game} python library that was created for
+this project.
+A detailed description of the library can be found in Section.
+% TODO: reference ambulance_game section {sec:ambulance_game_library}.
+The following code shows how the payoff matrices can be obtained using the
+\lstinline[style=pystyle]{ambulance_game} library.
+
+\begin{lstlisting}[style=pystyle]
+>>> import ambulance_game as abg
+>>> A, B, R = abg.game.get_payoff_matrices(
+...     lambda_2=lambda_2,
+...     lambda_1_1=lambda_1_A,
+...     lambda_1_2=lambda_1_B,
+...     mu_1=mu_A,
+...     mu_2=mu_B,
+...     num_of_servers_1=num_of_servers_A,
+...     num_of_servers_2=num_of_servers_B,
+...     system_capacity_1=system_capacity_A,
+...     system_capacity_2=system_capacity_B,
+...     buffer_capacity_1=buffer_capacity_A,
+...     buffer_capacity_2=buffer_capacity_B,
+...     target=target,
+...     alpha=alpha,
+...     p_hat=p_hat,
+... )
+>>> game = nash.Game(A, B)
+>>> game
+Bi matrix game with payoff matrices:
+<BLANKLINE>
+Row player:
+[[0.99934675 0.99934675]
+ [0.99282972 0.99828249]]
+<BLANKLINE>
+Column player:
+[[0.98725977 0.99408002]
+ [0.98725977 0.99312791]]
+
+\end{lstlisting}
+
+Having the payoff matrices, the Nash Equilibrium can be obtained using the
+support enumeration algorithm or the Lemke-Howson algorithm implemented in the
+\lstinline[style=pystyle]{nashpy} library.
+See Section~\ref{sec:game_intro_nash_equilibrium} for more details on these
+algorithms.
+
+\begin{lstlisting}[style=pystyle]
+>>> tuple(game.support_enumeration())
+((array([1., 0.]), array([0., 1.])),)
+>>> tuple(game.lemke_howson(initial_dropped_label=0))
+(array([1., 0.]), array([0., 1.]))
+
+\end{lstlisting}
+
+Both algorithms return the same set of star, which is for queueing system
+\(A\) to always choose a threshold of \(T_A = 1\) and for queueing system
+\(B\) to always choose a threshold of \(T_B = 2\).
+This means, that queueing system \(A\) will block type \(2\) individuals when
+there is \(1\) individual in node 1 and queueing system \(B\) will block type
+\(2\) individuals only when node \(1\) is at maximum capacity.
+
+In addition, the learning algorithms described in
+Section~\ref{sec:game_intro_learning_algorithms} can be used to obtain
```

Comment:
```suggestion
As the sizes of the queueing systems considered grow the strategy spaces will as well. This can increase the complexity of finding equilibria.
In addition, equilibria might not necessarily emerge, thus the learning algorithms described in
Section~\ref{sec:game_intro_learning_algorithms} can be used to obtain
```